modules/ocf_stats/manifests/prometheus.pp

```diff
@@ -1,8 +1,21 @@
 # prometheus daemon config
 class ocf_stats::prometheus {
+  file {
+    '/usr/local/bin/gen-prometheus-nodes':
+      source => 'puppet:///modules/ocf_stats/prometheus/gen-prometheus-nodes',
+      mode   => '0755';
+  }
+
+  cron { 'gen-prometheus-nodes':
+    command => '/usr/local/bin/gen-prometheus-nodes /var/local/prometheus-nodes.json',
+    user    => 'root',
+    minute  => '03',
+    require => File['/usr/local/bin/gen-prometheus-nodes'];
+  }
+
   class { '::prometheus':
     version        => '2.0.0',
-    extra_options  => '--web.listen-address="127.0.0.1:9090" --web.external-url=http://127.0.0.1/prom',
+    extra_options  => '--web.listen-address="127.0.0.1:9090" --web.external-url=http://127.0.0.1/prometheus --storage.tsdb.retention=2y',
     alerts         => {},
```

Comment:
Is this necessary?

Reply:
Retention is 14 days by default. After some discussion in the `#prometheus` IRC channel, I was told that it makes sense to retain more until we set up a proper long-term storage backend (which will happen in a seprate PR).